pkg/mcproducer/scripts/replace-kernel-module.sh

```diff
@@ -19,17 +21,36 @@ EOF
     cat "$kmm_config_file_filepath"
 }
 
-echo "before checking image tar file presence"
-if [ -e $kernel_module_image_filepath ]; then
-    echo "Image file $kernel_module_image_filepath found on the local file system, creating kmm config file"
+echo "before checking image presence"
+if [ -n "$(podman images -q $full_kernel_module_image 2> /dev/null)" ]; then
+    echo "Image $full_kernel_module_image found on the local file system, creating kmm config file"
     create_kmm_config
-    echo "running kernel-management worker image"
-    podman run --user=root --privileged -v /lib/modules:/lib/modules -v $kmm_config_file_filepath:/etc/kmm-worker/config.yaml -v $kernel_module_image_filepath:$kernel_module_image_filepath $worker_image kmod load --tarball /etc/kmm-worker/config.yaml
+    mkdir -p /tmp/opt/lib/modules
+    podman pod create --name kmm-pod
+    echo "creating init container"
+    podman create \
+      --pod kmm-pod \
+      --init-ctr=always \
+      -v /tmp/opt/lib/modules:/tmp/opt/lib/modules:Z \
```

Comment:
the shared volume between init-container and worker container should be the same on the host. So if you define that the shared volume on the host is /tmp/opt (i would rename opt to something else in order not to confuse with the opt in the container image), than worker container should also mount it

Reply:
The day2 solution is mounting an `EmptyDir` (regardless of where it is stored on the host) and mounting it to `/tmp/` in both, the init-container and the worker-container.

When we add firmware, we won't need to add another mount cause they will be looked up somewhere in `/tmp/` as well.